packages/mobile/src/screens/product-info/product-info.tsx

```diff
@@ -31,34 +39,70 @@ import {
 
 const ProductInfo: FC = () => {
   const { colors } = useCustomTheme();
+  const [isLoading, setIsLoading] = useState(false);
   const dispatch = useAppDispatch();
   const product = useAppSelector(selectProduct);
+  const user = useAppSelector(selectCurrentUser);
+  const favoriteIds = useAppSelector(selectFavoriteIds);
   const route =
     useRoute<
       RouteProp<Pick<RootNavigationParamList, RootScreenName.ITEM_INFO>>
     >();
   const id = route.params?.itemId;
+  const isFavorite = favoriteIds.includes(id);
 
   useEffect(() => {
     dispatch(productActions.loadProductInfo(id));
     dispatch(productActions.updateProductViews(id));
+    if (user) {
+      if (favoriteIds) {
+        favoriteIds.map((id) => dispatch(productsActions.addToFavorite(id)));
```

Comment:
???

Pls review all this logic. It looks messy

Reply:
fixed, some part moved to signIn handler